src/treeffuser/_base_tabular_diffusion.py

```diff
@@ -57,10 +83,28 @@ def get_new_score_model(self) -> ScoreModel:
         Return the score model.
         """
 
+    def _validate_data(
+        self,
+        X: Optional[ndarray] = None,
+        y: Optional[ndarray] = None,
+        validate_X: bool = True,
+        validate_y: bool = True,
+    ) -> Tuple[
+        Optional[Float[ndarray, "batch x_dim"]], Optional[Float[ndarray, "batch y_dim"]]
+    ]:
+        """Reshape X and/or y to adhere to the (batch, n_dim) convention, and cast them as flows."""
+        if validate_X and X is not None:
+            X = _check_array(X)
+
+        if validate_y and y is not None:
+            y = _check_array(y)
+
+        return X, y
+
     def fit(
         self,
-        X: Float[ndarray, "batch x_dim"],
-        y: Float[ndarray, "batch y_dim"],
+        X: ndarray,
+        y: ndarray,
         cat_idx: Optional[List[int]] = None,
     ):
         """
```

Comment:
I think docstring should be modified here to talk about the behavior with shapes